var/spack/repos/builtin/packages/amdlibm/package.py

```diff
@@ -41,29 +45,26 @@ class Amdlibm(SConsPackage):
     patch("0001-libm-ose-Scripts-cleanup-pyc-files.patch", when="@2.2")
     patch("0002-libm-ose-prevent-log-v3.c-from-building.patch", when="@2.2")
 
-    conflicts("%gcc@:9.1.0", msg="Minimum required GCC version is 9.2.0")
-    conflicts("%gcc@11.2.0:", msg="Maximum required GCC version is 11.1.0")
-    conflicts(
-        "%aocc@3.2.0",
-        when="@2.2:3.0",
-        msg="amdlibm 2.2 and 3.0 versions are not supported with AOCC 3.2.0",
-    )
+    conflicts("%gcc@:9.1.0", msg="Minimum supported GCC version is 9.2.0")
+    conflicts("%gcc@12.2.0:", msg="Maximum supported GCC version is 12.1.0")
+    conflicts("%clang@9:", msg="Minimum supported Clang version is 9.0.0")
+    conflicts("%aocc@3.2.0", msg="dependency on python@3.6.2")
 
-    def build_args(self, spec, prefix):
+    def build_args(self, *args, **kwargs):
```

Comment:
Why did you introduce this change? Can you please revert and use a more static signature signature?

Reply:
amdlibm builds are failing at the install stage with below error:

--snip--
~/spack/etc/spack/licenses/amdlibm/AMDLIBM_EULA.pdf" 25L, 842B written
    ==> amdlibm: Executing phase: 'build'
    ==> amdlibm: Executing phase: 'install'
    ==> Error: TypeError: Amdlibm.install_args() missing 2 required positional arguments: 'spec' and 'prefix'
    
    /home/amd-toolchain-support/spack/lib/spack/spack/builder.py:139, in __forward:
      >>    139        def __forward(self, *args, **kwargs):
            140            return self.__getattr__(fn_name)(*args, **kwargs)
--snip--

Reply:
Can you revert? The error seems to point to another function. Also, it's bad practice to have catch all signatures to work around errors when passing parameters. If there's still an error it means there's a bug somewhere in Spack we need to fix.